packages/ember-glimmer/lib/utils/iterable.js

```diff
@@ -115,16 +144,28 @@ class Iterable {
       return iterable.length > 0 ? new ArrayIterator(iterable, keyFor) : EMPTY_ITERATOR;
     } else if (isEmberArray(iterable)) {
       return new EmberArrayIterator(iterable, keyFor);
+    } else if (iterable instanceof Object) {
```

Comment:
This makes it possible to do `{{#each someObject}}` in general, which I don't think is currently supported. Can your AST transform add a marker here? Something like

`{{#each-in foo |bar baz|}}`

becomes

`{{#each (-each-in foo) |bar baz|}}`

`-each-in` can return a simple wrapper with a special symbol, and you can check for that symbol here instead of doing `instanceof Object`. (It also allows you to keep the order of the block params, which makes it easier to handle `|foo|` vs `|foo bar|`.)